tests/lib/c_lib/src/main.c

```diff
@@ -319,6 +322,321 @@ void test_bsearch(void)
 	zassert_not_null(result, "bsearch -key found");
 }
 
+/**
+ *
+ * @brief Test abs function
+ *
+ */
+
+void test_abs(void)
+{
+	int val = -5, value = 5;
+
+	zassert_equal(abs(val), 5, "abs -5");
+	zassert_equal(abs(value), 5, "abs 5");
+}
+
+/**
+ *
+ * @brief Test atoi function
+ *
+ */
+
+void test_atoi(void)
+{
+	char *str = "123";
+
+	zassert_equal(atoi(str), 123, "atoi error");
+}
+
+/**
+ *
+ * @brief Test isalnum function
+ *
+ */
+void test_isalnum(void)
+{
+	int str1 = '1',  str2 = 'a', str3 = '!';
+
+	zassert_true(isalnum(str1), "isalnum '1'");
+	zassert_true(isalnum(str2), "isalnum 'a'");
+	zassert_false(isalnum(str3), "isalnum '!'");
+}
+
+/**
+ *
+ * @brief Test isalpha function
+ *
+ */
+void test_isalpha(void)
+{
+	int ch1 = '1', ch2 = 'b';
+
+	zassert_false(isalpha(ch1), "isalpha '1'");
+	zassert_true(isalpha(ch2), "isalpha 'b'");
+}
+
+/**
+ *
+ * @brief Test isdigit function
+ *
+ */
+void test_isdigit(void)
+{
+	int ch1 = '1', ch2 = 'a';
+
+	zassert_true(isdigit(ch1), "isdigit '1'");
+	zassert_false(isdigit(ch2), "isdigit 'a'");
+}
+
+/**
+ *
+ * @brief Test isgraph function
+ *
+ */
+void test_isgraph(void)
+{
+	int ch = '1';
+	int sh = ' ';
+
+	zassert_true(isgraph(ch), "isgraph 1");
+	zassert_false(isgraph(sh), "isgraph ' '");
+}
+
+/**
+ *
+ * @brief Test isprint function
+ *
+ */
+void test_isprint(void)
+{
+	int ch = 0x7f, sh = '1';
+
+	zassert_false(isprint(ch), "isprint 0x7f");
+	zassert_true(isprint(sh), "isprint '1'");
+}
+
+/**
+ *
+ * @brief Test isspace function
+ *
+ */
+void test_isspace(void)
+{
+	int ch = 'a', sh = ' ';
+
+	zassert_false(isspace(ch), "isspace 'a'");
+	zassert_true(isspace(sh), "isspace ' '");
+}
+
+/**
+ *
+ * @brief Test isupper function
+ *
+ */
+void test_isupper(void)
+{
+	int ch = 'a', sh = 'A';
+
+	zassert_false(isupper(ch), "isupper 'a'");
+	zassert_true(isupper(sh), "isupper 'A'");
+}
+
+/**
+ *
+ * @brief Test isxdigit function
+ *
+ */
+void test_isxdigit(void)
+{
+	int ch1 = 'e', ch2 = 'z';
+
+	zassert_true(isxdigit(ch1), "isxdigit 'e'");
+	zassert_false(isxdigit(ch2), "isxdigit 'z'");
+}
+
+/**
+ *
+ * @brief Test menchr function
+ *
+ */
+static const char str[] = "testfunction";
+void test_memchr(void)
+{
+	int ch1 = 'e', ch2 = 'z';
+	char *ret;
+
+	ret = memchr(str, ch1, strlen(str));
+	zassert_not_null(ret, "memchr 'testfunction' 'e' ");
+
+	ret = memchr(str, ch2, strlen(str));
+	zassert_is_null(ret, "memchr 'testfunction' 'z'");
+}
+
+/**
+ *
+ * @brief Test memcpy function
+ *
+ */
+void test_memcpy(void)
+{
+	char arr[10];
+	char str[] = "12345";
+
+	memcpy(arr, str, sizeof(str));
+	zassert_true((memcmp(str, arr, sizeof(str))) == 0,
```

Comment:
Use `zassert_equal`

Reply:
Hi @stephanosio I think zassert_true and zassert_equal has same function in this part. I have no idea why need to change it, could you tell me the reason or the benefit?

Reply:
Let me ask you this:
1. why do you think we have `zassert_equal` to begin with?
2. why don't we convert everything that does not use `zassert_true` to use `zassert_true`? (because we CAN do that)

Reply:
Ok, I got your point. You meaning is don't detour if we have the best way, right?